src/Glpi/Asset/Asset.php

```diff
@@ -454,4 +543,49 @@ public function getCloneRelations(): array
         }
         return array_unique($relations);
     }
+
+    public static function getSpecificValueToSelect($field, $name = '', $values = '', array $options = [])
+    {
+        if ($options['searchopt']['datatype'] !== 'specific' || !isset($options['searchopt']['field_definition'])) {
+            return parent::getSpecificValueToSelect($field, $name, $values, $options);
+        }
+
+        // Handle complex custom fields
+        /** @var CustomField $field_definition */
+        $field_definition = $options['searchopt']['field_definition'];
+        return $field_definition->getFieldType()::getSpecificValueToSelect($field, $name, $values, $options);
+    }
+
+    public static function getSpecificValueToDisplay($field, $values, array $options = [])
+    {
+        if ($options['searchopt']['datatype'] !== 'specific' || !isset($options['searchopt']['field_definition'])) {
+            return parent::getSpecificValueToDisplay($field, $values, $options);
+        }
+
+        // Handle complex custom fields
+        /** @var CustomField $field_definition */
+        $field_definition = $options['searchopt']['field_definition'];
+        return $field_definition->getFieldType()::getSpecificValueToDisplay($field, $values, $options);
+
+        return parent::getSpecificValueToDisplay($field, $values, $options);
+    }
+
+    public static function addWhere($link, $nott, $itemtype, $ID, $searchtype, $val, $opt)
+    {
+        /** @var \DBmysql $DB */
+        global $DB;
+
+        if ($opt['datatype'] !== 'specific' || !isset($opt['field_definition'])) {
+            return '';
+        }
+        $field_definition = $opt['field_definition'];
+        if ($field_definition->fields['type'] === 'dropdown') {
+            $value_to_compare = $opt['computation'];
+            $operator = $nott ? '!=' : '=';
+
+            $val = $DB::quoteValue((int) $val);
+            return "$link ($val $operator $value_to_compare)";
+        }
+        return '';
+    }
```

Comment:
All of this is used for dropdown only, so I guess everything is working fine for other kind of fields. Is there any reason that prevent it for being done in the Search classes? These features are existing for any dropdown, and, unless I miss something, the only difference we have here is that we are defining a specific `computation` option.

@orthagh @AdrienClairembault @trasher your help is welcome on this point.

Reply:
I pushed a commit to remove all this override requirements. See https://github.com/glpi-project/glpi/pull/17462/commits/42ea9648086fb53da71c7f9a94a89e9cd3a14548.

I had to use some kind of a hack on the JOIN conditions:
```php
                        $nocondition_join = [
                            'LEFT JOIN' => [
                                "$new_table$AS" => [
                                    'ON' => [
                                        $rt => 'id',
                                        new QueryExpression("$rt.id"),
                                    ],
                                ]
                            ]
                        ];
```

I did not take time how I could handle it differently and not get an error from the iterator about the way JOIN should be declared.